Dockerfile.cross

```diff
@@ -0,0 +1,297 @@
+    --mount=type=cache,target=/root/.cache \
+    git fetch origin \
+    && git checkout -q "$ROOTLESSKIT_VERSION" \
+    && go build -o /out/rootlesskit -ldflags="$ROOTLESSKIT_LDFLAGS" -v ./cmd/rootlesskit \
+    && go build -o /out/rootlesskit-docker-proxy -ldflags="$ROOTLESSKIT_LDFLAGS" -v ./cmd/rootlesskit-docker-proxy
+
+FROM binary-dummy AS rootlesskit-darwin
+FROM binary-dummy AS rootlesskit-freebsd
+FROM rootlesskit-build AS rootlesskit-linux
+FROM binary-dummy AS rootlesskit-windows
+FROM rootlesskit-${TARGETOS} AS rootlesskit
+
+# containerutility
+FROM build-base-${BASE_VARIANT} AS containerutility-src
+WORKDIR /usr/src
+RUN git clone https://github.com/docker-archive/windows-container-utility.git containerutility
+
+FROM build-base-${BASE_VARIANT} AS containerutility-base-alpine
+ARG TARGETPLATFORM
+RUN xx-apk add --no-cache musl-dev gcc g++
+
+FROM build-base-${BASE_VARIANT} AS containerutility-base-bullseye
+ARG TARGETPLATFORM
+RUN xx-apt-get install -y binutils gcc g++ dpkg-dev pkg-config
+
+FROM containerutility-base-${BASE_VARIANT} AS containerutility-build
+ARG CONTAINERUTILITY_VERSION
+RUN --mount=from=containerutility-src,src=/usr/src/containerutility,rw \
+    --mount=type=cache,target=/root/.cache \
+    git fetch origin \
+    && git checkout -q "$CONTAINERUTILITY_VERSION" \
+    && CC="$(xx-info)-gcc" CXX="$(xx-info)-g++" make \
+    && mkdir /out \
+    && mv containerutility.exe /out/
+
+FROM binary-dummy AS containerutility-darwin
+FROM binary-dummy AS containerutility-freebsd
+FROM binary-dummy AS containerutility-linux
+FROM containerutility-build AS containerutility-windows-amd64
+FROM binary-dummy AS containerutility-windows-arm64
+FROM containerutility-windows-${TARGETARCH} AS containerutility-windows
+FROM containerutility-${TARGETOS} AS containerutility
+
+# vpnkit
+# TODO: build from source instead
+FROM scratch AS vpnkit-windows
+FROM scratch AS vpnkit-linux-386
+FROM djs55/vpnkit:${VPNKIT_VERSION} AS vpnkit-linux-amd64
+FROM scratch AS vpnkit-linux-arm
+FROM djs55/vpnkit:${VPNKIT_VERSION} AS vpnkit-linux-arm64
+FROM scratch AS vpnkit-linux-ppc64le
+FROM scratch AS vpnkit-linux-riscv64
+FROM scratch AS vpnkit-linux-s390x
+FROM vpnkit-linux-${TARGETARCH} AS vpnkit-linux
+FROM vpnkit-${TARGETOS} AS vpnkit
+
+FROM build-base-alpine AS build-alpine
+RUN apk add --no-cache lld
+ARG TARGETPLATFORM
+RUN xx-apk add --no-cache musl-dev gcc g++ pkgconf \
+    btrfs-progs-dev \
+    lvm2-dev \
+    libseccomp-dev \
+    libsecret-dev \
+    eudev-dev
+
+FROM build-base-bullseye AS build-bullseye
+RUN apt-get update && apt-get install -y lld
+ARG TARGETPLATFORM
+RUN xx-apt-get install --no-install-recommends -y binutils gcc g++ pkg-config dpkg-dev \
+    libapparmor-dev \
+    libbtrfs-dev \
+    libdevmapper-dev \
+    libseccomp-dev \
+    libsecret-1-dev \
+    libsystemd-dev \
+    libudev-dev
+
+FROM build-${BASE_VARIANT} AS build
+COPY --from=gowinres /out/ /usr/local/bin
+COPY --from=tini /out/ /out
```

Comment:
Why are these copied to the build stage. Should only by to binary stage.

Reply:
Because we want to generate the checksums file for them. Checksums fiel is created by `./hack/build/release`. Maybe we could find a better way and have a stage that does that job instead.

Reply:
Should not be in the same step that runs `go build`

Reply:
Moved in a dedicated stage